### PR TITLE
fix(metrics-api): Allow order by percentiles [INGEST-544] [INGEST-807]

### PR DESCRIPTION
**Pull request by jjbayer**
The limited order by functionality introduced in https://github.com/getsentry/sentry/pull/30091 did not yet allow ordering by one of the percentiles (`p50`, `p75`, ...). This PR fixes that and simplifies some code along the way.

### CODE REVIEW

File: src/sentry/snuba/metrics.py [hunk 1]
```diff
@@ -866,10 +846,9 @@ def _extract_data(self, entity, data, groups):
         for op in ops:
             key = f"{op}({metric_name})"
 
-            field = _OP_TO_FIELD[entity][op].snuba_alias
-            value = data[field]
-            if field == "percentiles":
-                value = value[Percentile[op].value]
+            value = data[op]
+            if op in _OPERATIONS_PERCENTILES:
+                value = value[0]
```
Comment (jjbayer) [inline src/sentry/snuba/metrics.py:851]: This would not be necessary if we could query `quantile(0.5)(value)` instead of `quantiles(0.5)(value)`, will follow up on that in the new year.

File: src/sentry/snuba/metrics.py [hunk 2]
```diff
@@ -1101,7 +1080,7 @@ def get_tag_values(
             value_ids = {value_id for ids in value_id_lists for value_id in ids}
 
         tags = [{"key": tag_name, "value": reverse_resolve(value_id)} for value_id in value_ids]
-        tags.sort(key=itemgetter("key"))
+        tags.sort(key=lambda tag: (tag["key"], tag["value"]))
```
Comment (jjbayer) [inline src/sentry/snuba/metrics.py:1083]: Unrelated fix.

File: src/sentry/snuba/metrics.py (hunk 1 above)
Comment (ahmedetefy) [reply src/sentry/snuba/metrics.py:851]: Can you elaborate on this?

**Review by ahmedetefy — APPROVED** [review]
